relay-server/src/services/health_check.rs

```diff
@@ -182,127 +196,74 @@ impl HealthCheckService {
         }
     }
 
-    async fn handle_is_healthy(&self, message: IsHealthy) -> Status {
-        let upstream = self.upstream_relay.clone();
-
+    async fn check_readiness(&mut self, shutdown: &ShutdownHandle) -> Status {
+        // Internal metric that we need to be logged in recurring intervals. This is a form of
+        // health check, but it does not contribute to the status of this service.
         if self.config.relay_mode() == RelayMode::Managed {
-            let fut = upstream.send(IsNetworkOutage);
-            tokio::spawn(async move {
-                if let Ok(is_outage) = fut.await {
-                    metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
-                }
-            });
-        }
-
-        if matches!(message, IsHealthy::Liveness) {
-            return Status::Healthy;
+            if let Ok(is_outage) = self.upstream_relay.send(IsNetworkOutage).await {
+                metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
+            }
         }
 
-        if self.is_shutting_down.load(Ordering::Relaxed) {
+        if shutdown.get().is_some() {
             return Status::Unhealthy;
         }
 
+        // System memory is sync and requires mutable access, but we still want to log errors.
+        let sys_mem = self.system_memory_probe();
+
         let (sys_mem, auth, agg, proj) = tokio::join!(
-            self.probe("system memory", self.system_memory_probe()),
+            self.probe("system memory", async { sys_mem }),
             self.probe("auth", self.auth_probe()),
             self.probe("aggregator", self.aggregator_probe()),
             self.probe("spool health", self.spool_health_probe()),
         );
 
         Status::from_iter([sys_mem, auth, agg, proj])
     }
-
-    async fn handle_message(&self, message: HealthCheck) {
-        let HealthCheck(message, sender) = message;
-
-        let ty = message.variant();
-        let response = relay_statsd::metric!(
-            timer(RelayTimers::HealthCheckDuration),
-            type = ty,
-            { self.handle_is_healthy(message).await }
-        );
-
-        sender.send(response);
-    }
 }
 
 impl Service for HealthCheckService {
     type Interface = HealthCheck;
 
-    fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
-        let service = Arc::new(self);
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
+        let (status_tx, status_rx) = watch::channel(Statuses::healthy());
+        let check_interval = self.config.health_sys_info_refresh_interval();
```

Comment:
Don't think we should attach this to the config option, the debounce was deliberate before but now we can just get rid of it and have a single delay and not tie the refresh time to the `health_sys` option.

Reply:
That's a good point. We still debounce memory checks, but now we debounce everything else with it. Per-se, I'd like to keep two separate configs (for us to tweak if necessary, not for customers to change), one for a check interval and one for a delay.

What if we rename + shim the option?